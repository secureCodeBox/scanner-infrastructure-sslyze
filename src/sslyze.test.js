/*
 *
 *  SecureCodeBox (SCB)
 *  Copyright 2015-2018 iteratec GmbH
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  	http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 * /
 */

const { worker } = require('./sslyze');
const sslscan = require('../lib/sslscan');
jest.mock('../lib/sslscan');

describe('sslyze', () => {
    describe('worker', () => {
        beforeEach(() => {
            sslscan.scanTarget.mockClear();
        });

        it('should work with a single target', async () => {
            const result = await worker([{ location: 'www.yahoo.com' }]);

            expect(sslscan.scanTarget).toBeCalledWith('www.yahoo.com', '');

            expect(result).toMatchSnapshot();
        });

        it('should take parameters', async () => {
            const result = await worker([
                { location: 'www.yahoo.com', attributes: { SSLYZE_PARAMETER: '' } },
            ]);

            expect(sslscan.scanTarget).toBeCalledWith('www.yahoo.com', '');

            expect(result).toMatchSnapshot();
        });

        it('should run multiple scans when multiple targets are configured', async () => {
            const result = await worker([
                { location: 'www.yahoo.com' },
                { location: 'www.yahoo.com', attributes: { SSLYZE_PARAMETER: '' } },
            ]);

            expect(sslscan.scanTarget).toHaveBeenCalledTimes(2);
            expect(sslscan.scanTarget).toBeCalledWith('www.yahoo.com', '');
            expect(sslscan.scanTarget).toBeCalledWith('www.yahoo.com', '');

            expect(result).toMatchSnapshot();
        });

        it('should throw an error if a scan fails', async () => {
            sslscan.scanTarget.mockReturnValueOnce(Promise.reject());

            expect(worker([{ location: 'localhost' }])).rejects.toThrowErrorMatchingSnapshot();

            expect(sslscan.scanTarget).toBeCalledWith('localhost', '');
        });
    });
});
