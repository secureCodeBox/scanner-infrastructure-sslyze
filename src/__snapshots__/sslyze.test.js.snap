// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sslyze worker should run multiple scans when multiple targets are configured 1`] = `
Object {
  "result": Array [
    Object {
      "attributes": Object {
        "cipher_suites": Array [
          "AES256-SHA",
          "AES128-GCM-SHA256",
          "AES128-SHA",
          "ECDHE-RSA-AES256-GCM-SHA384",
          "ECDHE-RSA-AES256-SHA384",
          "ECDHE-RSA-AES256-SHA",
          "ECDHE-RSA-AES128-GCM-SHA256",
          "ECDHE-RSA-AES128-SHA256",
          "ECDHE-RSA-AES128-SHA",
        ],
        "hostname": "www.securecodebox.io",
        "ip_address": "185.199.109.153",
        "port": 443,
        "tls_versions": Array [
          "TLS 1.2",
        ],
      },
      "category": "TLS Service Info",
      "description": "",
      "hint": null,
      "id": "49bf7fd3-8512-4d73-a28f-608e493cd726",
      "location": "www.securecodebox.io:443",
      "name": "TLS Service",
      "osi_layer": "PRESENTATION",
      "reference": null,
      "severity": "INFORMATIONAL",
    },
    Object {
      "attributes": Object {
        "cipher_suites": Array [
          "AES256-SHA",
          "AES128-GCM-SHA256",
          "AES128-SHA",
          "ECDHE-RSA-AES256-GCM-SHA384",
          "ECDHE-RSA-AES256-SHA384",
          "ECDHE-RSA-AES256-SHA",
          "ECDHE-RSA-AES128-GCM-SHA256",
          "ECDHE-RSA-AES128-SHA256",
          "ECDHE-RSA-AES128-SHA",
        ],
        "hostname": "www.securecodebox.io",
        "ip_address": "185.199.109.153",
        "port": 443,
        "tls_versions": Array [
          "TLS 1.2",
        ],
      },
      "category": "TLS Service Info",
      "description": "",
      "hint": null,
      "id": "49bf7fd3-8512-4d73-a28f-608e493cd726",
      "location": "www.securecodebox.io:443",
      "name": "TLS Service",
      "osi_layer": "PRESENTATION",
      "reference": null,
      "severity": "INFORMATIONAL",
    },
  ],
}
`;

exports[`sslyze worker should take parameters 1`] = `
Object {
  "result": Array [
    Object {
      "attributes": Object {
        "cipher_suites": Array [
          "AES256-SHA",
          "AES128-GCM-SHA256",
          "AES128-SHA",
          "ECDHE-RSA-AES256-GCM-SHA384",
          "ECDHE-RSA-AES256-SHA384",
          "ECDHE-RSA-AES256-SHA",
          "ECDHE-RSA-AES128-GCM-SHA256",
          "ECDHE-RSA-AES128-SHA256",
          "ECDHE-RSA-AES128-SHA",
        ],
        "hostname": "www.securecodebox.io",
        "ip_address": "185.199.109.153",
        "port": 443,
        "tls_versions": Array [
          "TLS 1.2",
        ],
      },
      "category": "TLS Service Info",
      "description": "",
      "hint": null,
      "id": "49bf7fd3-8512-4d73-a28f-608e493cd726",
      "location": "www.securecodebox.io:443",
      "name": "TLS Service",
      "osi_layer": "PRESENTATION",
      "reference": null,
      "severity": "INFORMATIONAL",
    },
  ],
}
`;

exports[`sslyze worker should throw an error if a scan fails 1`] = `"Failed to execute SSLyze scan."`;

exports[`sslyze worker should work with a single target 1`] = `
Object {
  "result": Array [
    Object {
      "attributes": Object {
        "cipher_suites": Array [
          "AES256-SHA",
          "AES128-GCM-SHA256",
          "AES128-SHA",
          "ECDHE-RSA-AES256-GCM-SHA384",
          "ECDHE-RSA-AES256-SHA384",
          "ECDHE-RSA-AES256-SHA",
          "ECDHE-RSA-AES128-GCM-SHA256",
          "ECDHE-RSA-AES128-SHA256",
          "ECDHE-RSA-AES128-SHA",
        ],
        "hostname": "www.securecodebox.io",
        "ip_address": "185.199.109.153",
        "port": 443,
        "tls_versions": Array [
          "TLS 1.2",
        ],
      },
      "category": "TLS Service Info",
      "description": "",
      "hint": null,
      "id": "49bf7fd3-8512-4d73-a28f-608e493cd726",
      "location": "www.securecodebox.io:443",
      "name": "TLS Service",
      "osi_layer": "PRESENTATION",
      "reference": null,
      "severity": "INFORMATIONAL",
    },
  ],
}
`;
